lib/Runtime/PlatformAgnostic/Platform/Windows/UnicodeText.cpp

```diff
@@ -0,0 +1,443 @@
+                Assert(SUCCEEDED(hr));
+                if (SUCCEEDED(hr))
+                {
+                    switch (category)
+                    {
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_LowercaseLetter:
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_UppercaseLetter:
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_TitlecaseLetter:
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_ModifierLetter:
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_OtherLetter:
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_LetterNumber:
+                            return UnicodeGeneralCategoryClass::CategoryClassLetter;
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_DecimalDigitNumber:
+                            return UnicodeGeneralCategoryClass::CategoryClassDigit;
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_LineSeparator:
+                            return UnicodeGeneralCategoryClass::CategoryClassLineSeparator;
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_ParagraphSeparator:
+                            return UnicodeGeneralCategoryClass::CategoryClassParagraphSeparator;
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_SpaceSeparator:
+                            return UnicodeGeneralCategoryClass::CategoryClassSpaceSeparator;
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_SpacingCombiningMark:
+                            return UnicodeGeneralCategoryClass::CategoryClassSpacingCombiningMark;
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_NonspacingMark:
+                            return UnicodeGeneralCategoryClass::CategoryClassNonSpacingMark;
+                        case UnicodeGeneralCategory::UnicodeGeneralCategory_ConnectorPunctuation:
+                            return UnicodeGeneralCategoryClass::CategoryClassConnectorPunctuation;
+                        default:
+                            break;
+                    }
+                }
+
+                return UnicodeGeneralCategoryClass::CategoryClassOther;
+            }, UnicodeGeneralCategoryClass::CategoryClassOther);
+        }
+
+        bool IsIdStart(codepoint_t codepoint)
+        {
+            return ExecuteWinGlobCodepointCheckApi(codepoint, &IUnicodeCharactersStatics::IsIdStart);
+        }
+
+        bool IsIdContinue(codepoint_t codepoint)
+        {
+            return ExecuteWinGlobCodepointCheckApi(codepoint, &IUnicodeCharactersStatics::IsIdContinue);
+        }
+
+        bool IsWhitespace(codepoint_t codepoint)
+        {
+            return ExecuteWinGlobCodepointCheckApi(codepoint, &IUnicodeCharactersStatics::IsWhitespace);
+        }
+
+        bool IsExternalUnicodeLibraryAvailable()
```

Comment:
Agreed that it would be good to make this standalone- will file issue to track this, need to move management of Windows.Globalization module lifetime outside of the thread context